lib/animina_web/potential_partner.ex

```diff
@@ -47,4 +50,35 @@ defmodule AniminaWeb.PotentialPartner do
     query
     |> Ash.Query.filter(gender: [eq: user.partner_gender])
   end
+
+  defp partner_geo_query(query, user) do
+    nearby_zip_codes =
+      get_nearby_cities(user.zip_code, user.search_range)
+      |> Enum.map(fn city -> city.zip_code end)
+
+    query
+    |> Ash.Query.filter(
+      or: [
+        zip_code: [in: nearby_zip_codes]
+      ]
+    )
+  end
+
+  # We use the haversine formula to get locations
+  # near the current city in the search range radius
+  defp get_nearby_cities(zip_code, search_range) do
+    current_city = City.by_zip_code!(zip_code)
+
+    City
+    |> Ash.Query.filter(
+      fragment(
+        "acos(sin(radians(?)) * sin(radians(lat)) + cos(radians(?)) * cos(radians(lat)) * cos(radians(?) - radians(lon))) * 6731 <= ?",
+        ^current_city.lat,
+        ^current_city.lat,
+        ^current_city.lon,
+        ^search_range
+      )
```

Comment:
Seeing GEO-related queries without an index on the database table always makes me nervous.
Wouldn't it make sense to let the database calculate distance?
Also see issue #190.